lib/Core/types.ts

```diff
@@ -1 +1,15 @@
 export type constVoid = () => void;
+
+/**
+ * Makes the specified properties of T optional.
+ *
+ * @example
+ * ```ts
+ *   Optional<Button, "id" | "name">
+ * ``
+ */
+export type Optional<T extends object, K extends keyof T = keyof T> = Omit<
+  T,
+  K
+> &
+  Partial<Pick<T, K>>;
```

Comment:
This file probs doesn't need to exist -  as we have [lib/Core/TypeModifiers.ts](https://github.com/TerriaJS/terriajs//blob/2b482eba0802e7f5f2100aeaa0d485187478aeed/lib/Core/TypeModifiers.ts)

Reply:
I missed that somehow. Moved `Optional` to TypeModifiers.ts